tutorials/editor_basics/working_with_text.md

```diff
@@ -1,208 +1,163 @@
 
-### 1.1.1 Creating a new action
-
-In this example we access the editor from an action.
-To create an action we need to extend the
-[AnAction.java](upsource:///platform/editor-ui-api/src/com/intellij/openapi/actionSystem/AnAction.java)
-class.
-
-
-```java
-public class EditorIllustration extends AnAction {
-}
-```
-
-
-### 1.1.2. Registering an action
-
-To register the action we should add the corresponding tag to the `<actions>` section of the plugin configuration file
-[plugin.xml](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/editor_basics/resources/META-INF/plugin.xml)
+![String replacement action](img/basics.png){:width="500px"}
 
+## Creating a New Menu Action
+In this example, we access the editor from an action.
+To review the fundamentals of creating and registering actions, refer to the [Actions Tutorial](/tutorials/action_system.md).
+The source code for the Java class in this example is [EditorIllustration](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/editor_basics/src/main/java/org/intellij/sdk/editor/EditorIllustration.java).
 
+To register the action, we must add the corresponding elements to the `<actions>` section of the plugin configuration file
+[plugin.xml](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/editor_basics/src/main/resources/META-INF/plugin.xml).
+The `EditorIllustration` action is registered in the group `EditorPopupMenu` so it will be available from the context menu when focus is on the editor:
 ```xml
-<actions>
-    <action id="EditorBasics.EditorIllustration" class="EditorIllustration" text="Editor Basics"
-        description="Illustrates how to plug an action in">
-        <add-to-group group-id="EditorPopupMenu" anchor="last"/>
+    <action id="EditorBasics.EditorIllustration"
+            class="org.intellij.sdk.editor.EditorIllustration"
+            text="Editor Replace Text"
+            description="Replaces selected text with 'Replacement'."
+            icon="EditorBasicsIcons.Sdk_default_icon">
+      <add-to-group group-id="EditorPopupMenu" anchor="first"/>
     </action>
-</actions>
 ```
 
-If an action is registered in the group `EditorPopupMenu`, like the sample above shows,
-it will be available from the context menu when the focus is located in the editor.
-
-
-### 1.1.3. Defining action's visibility
-
-To determine conditions by which the action will be visible and available for being executed we need to override its
-`public void update(AnActionEvent e)` method.
-
+## Defining the Menu Action's Visibility
+To determine conditions by which the action will be visible and available requires `EditorIllustration` to override the
+`AnAction.update()` method.
 ```java
 public class EditorIllustration extends AnAction {
+    @Override                                        
```

Comment:
omit this sample code, this should be covered in "Actions tutorial"

Reply:
Done.